platform/darwin/src/MGLStyle.mm

```diff
@@ -142,12 +175,66 @@ - (void)insertLayer:(id <MGLStyleLayer, MGLStyleLayer_Private>)styleLayer
 
 - (void)addSource:(MGLSource *)source
 {
-    self.mapView.mbglMap->addSource(std::move([source mbgl_source]));
+    self.mapView.mbglMap->addSource([source mbgl_source]);
 }
 
 - (void)removeSource:(MGLSource *)source
 {
     self.mapView.mbglMap->removeSource(source.sourceIdentifier.UTF8String);
 }
 
+- (NS_ARRAY_OF(NSString *) *)styleClasses
+{
+    NSMutableArray *returnArray = [NSMutableArray array];
```

Comment:
Not critical, but you could initialize this array with a capacity of `appliedClasses.size()`.